libraries/AP_Notify/AP_Notify.h

```diff
@@ -111,10 +122,17 @@ class AP_Notify
     static const struct AP_Param::GroupInfo var_info[];
 
     bool buzzer_enabled() const { return _buzzer_enable; }
+
+    static void send_text(const char *str) { snprintf(_send_text, sizeof(_send_text), "%s", str); }
```

Comment:
>Also send_text() just prints string to the buffer.

I agree that snprintf is the safest, but it *extremely* slow. It's not a simple mem copy, it's a string parsing operation. Even though it's a relatively simple one it's still much more complicated than a simple strncpy and I think it's safe to say that we will always have a null terminated string arrive here. Even not, you're clipping to the _send_text size so it's safe. I propose either using strcpy() and/or check if _type != DISPLAY_OFF.

See the comparison here: http://stackoverflow.com/questions/12275381/strncpy-vs-sprintf

Execution time, including parameter shuffling on/off call stack:

strcpy   43 instructions
sprintf  467 instructions
Program/ROM space allocated:

strcpy   56 bytes
sprintf  1488 bytes
RAM/stack space allocated:

strcpy   0 bytes
sprintf  15 bytes